src/advanced-concepts/scalability/_mo/main3.mo

```diff
@@ -0,0 +1,33 @@
+// ANCHOR: a
+// `main.mo`
+import Principal "mo:base/Principal";
+import Buffer "mo:base/Buffer";
+import Error "mo:base/Error";
+
+import User "actor-class";
+
+actor {
+    let users = Buffer.Buffer<User.User>(1);
+
+    public func newUser(name : Text) : async Principal {
+        let instance = await User.User(name);
```

Comment:
On the IC, you'd actually need to provide some cycles to the call of the actor class constructor, to pay for the new canister creation. On the playground, this may just work without adding cycles. Maybe worth adding a comment.